services/web/client/source/class/qxapp/component/widget/FilePicker.js

```diff
@@ -88,52 +82,59 @@ qx.Class.define("qxapp.component.widget.FilePicker", {
   members: {
     __tree: null,
     __selectBtn: null,
-    __currentUserId: "ODEI-UUID",
 
     _createChildControlImpl: function(id) {
       let control;
       switch (id) {
-        case "addButton":
-          control = new qx.ui.form.Button(this.tr("Add file(s)"));
-          this._add(control);
-          break;
-        case "treeMenu":
-          control = new qx.ui.tree.VirtualTree(null, "label", "children").set({
-            openMode: "none"
-          });
+        case "filesTree":
+          control = new qxapp.component.widget.FilesTree();
           this._add(control, {
             flex: 1
           });
           break;
-        case "selectButton":
-          control = new qx.ui.form.Button(this.tr("Select"));
-          this._add(control);
-          break;
         case "progressBox":
           control = new qx.ui.container.Composite(new qx.ui.layout.HBox());
           this._addAt(control, 1);
           break;
+        case "addButton":
+          control = new qx.ui.form.Button(this.tr("Add file(s)"));
+          this._add(control);
+          break;
+        case "selectButton":
+          control = new qx.ui.form.Button(this.tr("Select"));
+          this._add(control);
+          break;
       }
 
       return control || this.base(arguments, id);
     },
 
-    buildTree: function() {
-      this.__getFiles();
+    __initResources: function() {
+      this.__tree.populateTree();
     },
 
-    __getFiles: function() {
-      let filesTreePopulator = new qxapp.utils.FilesTreePopulator(this.__tree);
-      filesTreePopulator.populateMyDocuments();
+    __selectionChanged: function() {
+      const data = this.__tree.getSelectedFile();
+      if (data) {
+        const isFile = data["isFile"];
+        this.__selectBtn.setEnabled(isFile);
+      } else {
+        this.__selectBtn.setEnabled(false);
```

Comment:
wouldn't this work as well?
this.__selectBtn.setEnabled(data && data["isFile"]);

Reply:
getSelectedFile() may return null
this.__selectBtn.setEnabled(data ? data["isFile"] : false);
would be the one-line solution

Reply:
preferable, oder?

Reply:
in C++ (data && data["isFile"]) would first evaluate data and be false in case of null.